README.md

```diff
@@ -1,33 +1,230 @@
 # swap-swear-and-swindle-suite
 
-This repository include a basic example of a courtroom contracts game.
+This repository include a basic abstract example of a courtroom contracts game.
 
-Contracts and tests are under contract directory.
-Though I tries to keep the contracts generic as possible ,the "reflector" game as describe below is currently part of the courtroom contract (courtroom.sol). On later phase it will be changes.
 
-## reflector game
-A very simple basic game and see how its fits to the defined swindle game and test various scenarios
-We can do very simple game as followed
+## Courtroom structure
+The courtroom suite includes :
+ - an abstract generic contract "SwearGame" which in a sense conduct the trial: iterate over the specific game witnesses ,trial statuses and verdict accordingly .
+ - a specific game rules contract which defined the trial rules such as transitions statuses,grace periods per state..  .(in our example it is "MirrorRules")
+ - A specific game witnesses contracts which in a sense are "expert witnesses" which can validate submitted evidences.  
```

Comment:
"It might be an idea to have an initial section that outlines the core primitives that are used throughout, to avoid any confusion over terminology."
It is currently below .. will add this initial section with the terminology .

" how the witnesses are meant to be employed and then transitioned between."
The transitions state are defined at the TrialRules contract.  It include a transition map from state to state. 
Yes. It is still missing the default witness though I still unsure under which scenario  this default witness will be called to testimony.